wallet/models.py

```diff
@@ -13,14 +17,21 @@ def __str__(self):
         return self.name
 
 
+@receiver(pre_save, sender=Category)
+def validate_unique(sender, instance, **kwargs):
+    # unique_together of parent, name and owner
+    if Category.objects.filter(owner=instance.owner, name=instance.name).exists():
+        raise ValidationError("Duplicate Category")
+
+
 def get_sentinel_category():
-    return Category.objects.get_or_create(name='default')[0]
+    return Category.objects.get_or_create(name='Default')[0]
 
 
 class Expense(models.Model):
     owner = models.ForeignKey(to=User, on_delete=models.CASCADE)
-    amount = models.FloatField()
+    amount = models.DecimalField(max_digits=10, decimal_places=2)
     category = models.ForeignKey(Category, on_delete=models.SET(get_sentinel_category))
```

Comment:
What if we let it be null and ask the client to display "Default"? Would that simplify the logic?

Reply:
they are just the same meaning nothing simpler